src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,189 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+
+    private static Ui ui;
+    private static TaskList tasklist;
+    private Storage storage;
+
+    public Parser(Ui ui, TaskList tasklist, Storage storage) {
+        this.ui = ui;
+        this.tasklist = tasklist;
+        this.storage = storage;
+    }
+
+    /**
+     * Parses user input to determine what Duke method to call.
+     *
+     * @param userInput input entered by user.
+     * @throws DukeException If user fails to specify what task they want to do.
+     */
+    public void parse(String userInput) throws DukeException {
+        if (userInput.equals("list")) {
+            showList();
+        } else if (userInput.length() > 4 && (userInput.substring(0, 4)).equals("mark")) {
+            markDone(userInput);
+            storage.save(tasklist);
+        } else if (userInput.length() > 6 && (userInput.substring(0, 6)).equals("unmark")) {
+            markUndone(userInput);
+            storage.save(tasklist);
+        } else if (userInput.length() > 4 && (userInput.substring(0, 4)).equals("todo")) {
+            addToDo(userInput);
+            storage.save(tasklist);
+        } else if (userInput.equals("todo")) {
+            throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+        } else if (userInput.equals("deadline")) {
+            throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+        } else if (userInput.equals("event")) {
+            throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+        } else if (userInput.length() > 8 && (userInput.substring(0, 8)).equals("deadline")) {
```

Comment:
This branch will match the input `deadlinesNextWeek`. Perhaps use the technique you discovered on line 54?

```suggestion
        } else if (userInput.startsWith("deadline ")) {
```